golang/lib/services/container_config.go

```diff
@@ -0,0 +1,138 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+package services
+
+const (
+	defaultKurtosisVolumeMountpoint = "/kurtosis-enclave-data"
+)
+
+type StaticFileID string
+
+// The ID of an artifact containing files that should be mounted into a service container
+type FilesArtifactID string
+
+// ====================================================================================================
+//                                    Config Object
+// ====================================================================================================
+// Docs available at https://docs.kurtosistech.com/kurtosis-client/lib-documentation
+type ContainerConfig struct {
+	image                        string
+	usedPortsSet                 map[string]bool
+	kurtosisVolumeMountpoint     string // Technically the enclave data volume, but we call it this for simplicity for the user
+	filesArtifactMountpoints     map[FilesArtifactID]string
+	entrypointOverrideArgs       []string
+	cmdOverrideArgs              []string
+	environmentVariableOverrides map[string]string
+}
+
+func (config *ContainerConfig) GetImage() string {
+	return config.image
+}
+
+func (config *ContainerConfig) GetUsedPortsSet() map[string]bool {
+	return config.usedPortsSet
+}
+
+func (config *ContainerConfig) GetKurtosisVolumeMountpoint() string {
+	return config.kurtosisVolumeMountpoint
+}
+
+func (config *ContainerConfig) GetFilesArtifactMountpoints() map[FilesArtifactID]string {
+	return config.filesArtifactMountpoints
+}
+
+func (config *ContainerConfig) GetEntrypointOverrideArgs() []string {
+	return config.entrypointOverrideArgs
+}
+
+func (config *ContainerConfig) GetCmdOverrideArgs() []string {
+	return config.cmdOverrideArgs
+}
+
+func (config *ContainerConfig) GetEnvironmentVariableOverrides() map[string]string {
+	return config.environmentVariableOverrides
+}
+
+// ====================================================================================================
+//                                      Builder
+// ====================================================================================================
+// TODO Defensive copies on all these With... functions???
+// Docs available at https://docs.kurtosistech.com/kurtosis-client/lib-documentation
+type ContainerConfigBuilder struct {
+	image                        string
+	usedPortsSet                 map[string]bool
+	kurtosisVolumeMountpoint     string
+	filesArtifactMountpoints     map[FilesArtifactID]string
+	entrypointOverrideArgs       []string
+	cmdOverrideArgs              []string
+	environmentVariableOverrides map[string]string
+}
+
+func NewContainerConfigBuilder(image string) *ContainerConfigBuilder {
+	return &ContainerConfigBuilder{
+		image:                        image,
+		usedPortsSet:                 map[string]bool{},
+		kurtosisVolumeMountpoint:     "",
```

Comment:
this is a bug! this should be `defaultKurtosisVolumeMountpoint`!

Reply:
uh! yes! you are right! good catch